shoop/front/template_helpers/product.py

```diff
@@ -44,28 +38,46 @@ def is_visible(context, product):
 
 
 @contextfunction
+@warn_about_renamed_method("", "get_product_cross_sells", "get_related_products", DeprecationWarning)
 def get_product_cross_sells(context, product, relation_type="related", count=4):
-    request = context["request"]
-    rtype = ProductCrossSellType.RELATED
-    if relation_type == "computed":
-        rtype = ProductCrossSellType.COMPUTED
-    elif relation_type == "recommended":
-        rtype = ProductCrossSellType.RECOMMENDED
+    return get_related_products(context, product, relation_type, count)
 
-    related_product_ids = list((
-        ProductCrossSell.objects
-        .filter(product1=product, type=rtype)
-        .order_by("weight")[:(count * 4)]).values_list("product2_id", flat=True)
-    )
 
-    related_products = list(
-        Product.objects
-        .filter(id__in=related_product_ids)
-        .list_visible(shop=request.shop, customer=request.customer)
-    )
+@contextfunction
+def get_related_products(context, product, relation_type="related", count=4):
+    """
+    Get at most `count` visible products related (by type `relation_type`) to the given `product`.
 
-    # Order related products by weight. Related product ids is in weight order.
-    # If same related product is linked twice to product then lowest weight stands.
-    related_products.sort(key=lambda prod: list(related_product_ids).index(prod.id))
+    :param context: The rendering context. This parameter is passed implicitly by Jinja2.
+    :type context: jinja2.runtime.Context
+    :param product: Source product
+    :type product: shoop.core.models.Product
+    :param relation_type: Relation type (string or ProductCrossSellType value)
+    :type relation_type: str|ProductCrossSellType
+    :param count: Number of products to return.
+    :type count: int
+    :return: List of at most `count` products.
+    :rtype: list[shoop.core.models.Product]
+    """
+    relation_type = map_enum(ProductCrossSellType, relation_type, default=ProductCrossSellType.RELATED)
```

Comment:
It would be more useful (for the template developer too) to raise an exception rather than returning possibly incorrect set of products (if e.g. the template developer typoes "related" as "reltaed"). Don't you think?